aws-xray-recorder-sdk-core/src/test/java/com/amazonaws/xray/contexts/LambdaSegmentContextTest.java

```diff
@@ -59,55 +58,71 @@ public void setupAWSXRay() {
     }
 
     @Test
-    public void testBeginSubsegmentWithNullTraceHeaderEnvironmentVariableResultsInADummySegmentParent() {
-        testMockContext(TraceHeader.fromString(null), FacadeSegment.class);
+    void testBeginSubsegmentWithNullTraceHeaderEnvironmentVariableResultsInADummySegmentParent() {
+        testContextResultsInFacadeSegmentParent();
     }
 
     @Test
-    public void testBeginSubsegmentWithIncompleteTraceHeaderEnvironmentVariableResultsInADummySegmentParent() {
-        testMockContext(TraceHeader.fromString("a"), FacadeSegment.class);
+    @SetEnvironmentVariable(key = "_X_AMZN_TRACE_ID", value = "a")
+    void testBeginSubsegmentWithIncompleteTraceHeaderEnvironmentVariableResultsInADummySegmentParent() {
+        testContextResultsInFacadeSegmentParent();
     }
 
     @Test
-    public void testBeginSubsegmentWithCompleteTraceHeaderEnvironmentVariableResultsInAFacadeSegmentParent() {
-        testMockContext(TraceHeader.fromString(TRACE_HEADER), FacadeSegment.class);
+    @SetEnvironmentVariable(key = "_X_AMZN_TRACE_ID", value = TRACE_HEADER)
+    void testBeginSubsegmentWithCompleteTraceHeaderEnvironmentVariableResultsInAFacadeSegmentParent() {
+        testContextResultsInFacadeSegmentParent();
     }
 
     @Test
-    public void testBeginSubsegmentWithCompleteButMalformedTraceHeaderEnvironmentVariableResultsInAFacadeSegmentParent() {
-        testMockContext(TraceHeader.fromString(MALFORMED_TRACE_HEADER), FacadeSegment.class);
+    @SetEnvironmentVariable(key = "_X_AMZN_TRACE_ID", value = MALFORMED_TRACE_HEADER)
+    void testBeginSubsegmentWithCompleteButMalformedTraceHeaderEnvironmentVariableResultsInAFacadeSegmentParent() {
+        testContextResultsInFacadeSegmentParent();
     }
 
     @Test
-    public void testLeakedSubsegmentsAreCleanedBetweenInvocations() {
-
+    @SetEnvironmentVariable(key = "_X_AMZN_TRACE_ID", value = TRACE_HEADER_2)
+    void testNotSampledSetsParentToSubsegment() {
```

Comment:
This is the new test, the rest are refactoring to replace powermock with junit5 extension